webapp/src/components/expanded_view/index.ts

```diff
@@ -34,27 +58,43 @@ const mapStateToProps = (state: GlobalState) => {
 
     let connectedDMUser;
     if (channel && isDMChannel(channel)) {
-        const otherID = getUserIdFromDM(channel.name, getCurrentUserId(state));
+        const otherID = getUserIdFromDM(channel.name, currentUserID);
         connectedDMUser = getUser(state, otherID);
     }
 
+    const {channelURL, channelDisplayName} = getChannelUrlAndDisplayName(state, channel);
+
+    const thread = getThread(state, threadID);
+
     return {
         show: expandedView(state),
-        currentUserID: getCurrentUserId(state),
+        currentUserID,
+        currentTeamID,
         profiles,
         pictures,
         statuses,
         callStartAt: voiceChannelCallStartAt(state, channel?.id) || 0,
         screenSharingID,
         channel,
+        channelTeam,
+        channelURL,
+        channelDisplayName,
         connectedDMUser,
+        threadID,
+        threadUnreadReplies: thread?.unread_replies,
+        threadUnreadMentions: thread?.unread_mentions,
+        rhsSelectedThreadID: getRhsSelectedPostId?.(state),
+        isRhsOpen: getIsRhsOpen?.(state),
     };
 };
 
-const mapDispatchToProps = (dispatch: Dispatch) => bindActionCreators({
+const mapDispatchToProps = {
```

Comment:
just curious why we don't need the `bindActionCreators`? (I'm not super familiar with it in the first place)

Reply:
You can pass a map of action creators and `connect` will bind them automatically. IMO it's easier than importing all the utilities and typing, but 0/5 on which one. (It's been used somewhat sparingly in the past, and [Harrison touched on it recently](https://community.mattermost.com/core/pl/ezcbewbadtbf7damzwiujnupcr).)